src/Dapr.Client/InvokeHttpClient.cs

```diff
@@ -0,0 +1,89 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+namespace Dapr
+{
+    using System;
+    using System.IO;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Text.Json;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using static Dapr.DaprUris;
+
+    /// <summary>
+    /// A client for interacting with the Dapr invoke endpoints using <see cref="HttpClient" />.
+    /// </summary>
+    public class InvokeHttpClient : InvokeClient
+    {
+        private readonly HttpClient client;
+        private readonly JsonSerializerOptions serializerOptions;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="InvokeHttpClient"/> class.
+        /// </summary>
+        /// <param name="client">The <see cref="HttpClient" />.</param>
+        /// <param name="serializerOptions">The <see cref="JsonSerializerOptions" />.</param>
+        public InvokeHttpClient(HttpClient client, JsonSerializerOptions serializerOptions = null)
+        {
+            if (client is null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+
+            this.client = client;
+            this.serializerOptions = serializerOptions;
+        }
+
+        /// <summary>
+        /// Invokes a method using the Dapr invoke endpoints using the properties supplied in the  <paramref name="envelope" /> variable.
+        /// </summary>
+        /// <param name="envelope">The envelope containing the invoke request parameters.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken" /> that can be used to cancel the operation.</param>
+        /// <typeparam name="TValue">The data type.</typeparam>
+        /// <returns>A <see cref="Task" /> that will return a deserialized object of the reponse from the Invoke request.</returns>
+        public override async Task<TValue> InvokeMethodAsync<TValue>(InvokeEnvelope envelope, CancellationToken cancellationToken = default)
```

Comment:
I think, passing three arguments here, without an InvokeEnvelope should be fine.
The api should handle all 3 types of invokes:
1. No Data and no return type. 
2. No Data but a return type.
3. Data but no return type.

Reply:
Hey Aman, I've removed the InvokeEnvelope and added the three separate methods for the different variations as you describe above. I wanted to be explicit with the three different options, hence the three separate methods with broadly the same implementation